tests/testTransform.py

```diff
@@ -266,17 +272,75 @@ def testInterface(self):
-    return unittest.TestSuite(suites)
+class CoaddTransformTestCase(utilsTests.TestCase):
+    """Check that CoaddSrcTransformTask is set up properly.
+
+    RunTransformTestCase, above, has tested the basic RunTransformTask mechanism.
+    Here, we just check that it is appropriately adapted for coadds.
+    """
+    MEASUREMENT_CONFIG_DATASET = "measureCoaddSources_config"
+
+    # The following are hard-coded in lsst.pipe.tasks.multiBand:
+    SCHEMA_SUFFIX = "Coadd_meas_schema"
+    SOURCE_SUFFIX = "Coadd_meas"
+    CALEXP_SUFFIX = "Coadd_calexp"
+
+    def setUp(self):
+        # We need a temporary repository in which we can store test configs.
+        self.repo = tempfile.mkdtemp()
+        with open(os.path.join(self.repo, "_mapper"), "w") as f:
+            f.write("lsst.obs.test.TestMapper")
+        self.butler = dafPersist.Butler(self.repo)
+
+        # Persist a coadd measurement config.
+        # We disable all measurement plugins so that there's no actual work
+        # for the TransformTask to do.
+        measCfg = MeasureMergedCoaddSourcesConfig()
+        measCfg.measurement.plugins.names = []
+        self.butler.put(measCfg, self.MEASUREMENT_CONFIG_DATASET)
+
+        # The record the type of coadd on which our supposed measurements have
+        # been carried out: we need to check this was propagated to the
+        # transformation task.
+        self.coaddName = measCfg.coaddName
+
+        # Since we disabled all measurement plugins, our catalog can be
+        # simple.
+        c = afwTable.SourceCatalog(afwTable.SourceTable.makeMinimalSchema())
+        self.butler.put(c, self.coaddName + self.SCHEMA_SUFFIX)
+
+        # Our transformation config needs to know the type of the measurement
+        # configuration.
+        trCfg = RunTransformConfig()
+        trCfg.inputConfigType = self.MEASUREMENT_CONFIG_DATASET
 
+        self.transformTask = CoaddSrcTransformTask(config=trCfg, log=None, butler=self.butler)
 
-def run(exit=False):
-    utilsTests.run(suite(), exit)
+    def tearDown(self):
+        del self.butler
+        del self.transformTask
+        shutil.rmtree(self.repo)
+
+    def testCoaddName(self):
+        """Check that we have correctly derived the coadd name."""
+        self.assertEqual(self.transformTask.coaddName, self.coaddName)
+
+    def testSourceType(self):
+        """Check that we have correctly derived the type of the measured sources."""
+        self.assertEqual(self.transformTask.sourceType, self.coaddName + self.SOURCE_SUFFIX)
+
+    def testCalexpType(self):
+        """Check that we have correctly derived the type of the measurement images."""
+        self.assertEqual(self.transformTask.calexpType, self.coaddName + self.CALEXP_SUFFIX)
+
+
+class MemoryTestCase(utilsTests.MemoryTestCase):
+    pass
+
+
+def setup_module(module):
+    utilsTests.init()
 
 if __name__ == "__main__":
-    run(True)
+    setup_module(sys.modules[__name__])
```

Comment:
I don't think that this line is necessary. It is my understanding that this block will be removed once the conversion to pytest is complete but the dependence on the unused "sys" package might remain. I would recommend either moving `import sys` to this block or just calling `utilsTests.init()` directly.